From 8fdf5b55515cb05f316bf8b676b0becb84334964 Mon Sep 17 00:00:00 2001
From: Ian McInerney <i.mcinerney17@imperial.ac.uk>
Date: Tue, 21 May 2024 16:45:23 +0100
Subject: [PATCH] Set import library for Clipper2 target when using system
 version

On Windows, the import library is also needed to link against the
Clipper2 DLL, and that must be specified using a different target
property.
---
 manifoldDeps.cmake | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/manifoldDeps.cmake b/manifoldDeps.cmake
index 194b8a1..d36e872 100644
--- a/manifoldDeps.cmake
+++ b/manifoldDeps.cmake
@@ -14,6 +14,10 @@ if(Clipper2_FOUND)
     add_library(Clipper2 SHARED IMPORTED)
     set_property(TARGET Clipper2 PROPERTY
         IMPORTED_LOCATION ${Clipper2_LINK_LIBRARIES})
+    if(WIN32)
+        set_property(TARGET Clipper2 PROPERTY
+            IMPORTED_IMPLIB ${Clipper2_LINK_LIBRARIES})
+    endif()
     target_include_directories(Clipper2 INTERFACE ${Clipper2_INCLUDE_DIRS})
 else()
     message(STATUS "clipper2 not found, downloading from source")
-- 
2.45.0

